commit 156f5a04185a789c8117b2e506b1e5d604d5d74c (refs/changes/94/2494/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-05T11:18:11-07:00 (2 years, 1 month ago)
    
    TOOLS-1852 sdc-clients: Fix cnapi test to run even if there is a down headnode

diff --git a/test/cnapi.test.js b/test/cnapi.test.js
index c7d28c3..151eb1d 100644
--- a/test/cnapi.test.js
+++ b/test/cnapi.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 var bunyan = require('bunyan');
@@ -150,7 +150,7 @@ test('cnapi', function (tt) {
                 minTimeout: 1000
             },
             log: new bunyan.createLogger({
-                name: 'cnapi_unit_test',
+                name: 'nodesdcclientstest-cnapi',
                 stream: process.stderr,
                 level: (process.env.LOG_LEVEL || 'info'),
                 serializers: bunyan.stdSerializers
@@ -161,9 +161,13 @@ test('cnapi', function (tt) {
 
     tt.test(' list servers', function (t) {
         cnapi.listServers({ headnode: true }, function (err, servers) {
-            t.ifError(err);
-            t.ok(servers);
-            SERVER = servers[0].uuid;
+            t.ifError(err, err);
+
+            // Choose a running server for subsequent tests.
+            SERVER = servers.filter(function (s) {
+                return (s.status === 'running');
+            })[0].uuid;
+
             t.end();
         });
     });
